roles/etcd/tasks/directory_permissions.yml

```diff
@@ -0,0 +1,50 @@
+---
+- name: directory permissions | check data permissions
+  shell: |
+    set -o pipefail && find {{ etcd_data_dir }} | xargs stat -c %U | grep -v etcd | wc -l
+  args:
+    executable: /bin/bash
+  register: etcd_permission
+  changed_when: false
+  when: etcd_deployment_type == "host"
+  failed_when: etcd_permission.rc > 2
+
+- name: directory permissions | check service
+  shell: |
+    set -o pipefail && pgrep -u etcd | wc -l
+  args:
+    executable: /bin/bash
+  register: etcd_service
+  changed_when: false
+  when: etcd_deployment_type == "host"
+  failed_when: etcd_service.rc > 2
+
+- name: directory permissions | stop etcd
+  service:
+    name: etcd
+    state: stopped
+    enabled: yes
+  when:
+    - etcd_deployment_type == "host"
+    - etcd_permission.stdout != "0" or etcd_service.stdout == "0"
+
+- name: directory permissions | set data permissions
+  file:
+    path: "{{ etcd_data_dir }}"
+    state: directory
+    owner: etcd
+    group: etcd
+    mode: 0700
```

Comment:
I still don't think we should change file permissions to 700, only ownership is enough IMO, and maybe permission of the top directory

Reply:
Well again, that's a security recommendation given by the CIS so as we are changing owernship let's also change mode, m2c at least

Reply:
I doubt CIS recommend to make data file executable ...

Reply:
indeed 7 not needed, I was trying to say no more groups/other perm 😄

Reply:
"mode: 0700" recomended for kube-bench

Reply:
For the top directory only ? or for all the files ? please link to the CIS kube-bench text, right now you are using recurse true.
I would split this in 2: for the top directory, `chown etcd:etcd / chmod 0700`, for everything in the directory only do `chown etcd:etcd`

Reply:
https://github.com/aquasecurity/kube-bench/blob/main/cfg/cis-1.6/master.yaml
 1.1.11
you're right